pkg/controller/vitessshard/reconcile_backup_job.go

```diff
@@ -219,31 +220,35 @@ func vtbackupSpec(key client.ObjectKey, vts *planetscalev2.VitessShard, parentLa
 	minRetentionTime := time.Duration(0)
 	minRetentionCount := 1
 
+	// Fill in the parts of a vttablet spec that make sense for vtbackup.
+	tabletSpec := &vttablet.Spec{
+		GlobalLockserver:         vts.Spec.GlobalLockserver,
+		Labels:                   labels,
+		Images:                   vts.Spec.Images,
+		KeyRange:                 vts.Spec.KeyRange,
+		Vttablet:                 &pool.Vttablet,
+		Mysqld:                   pool.Mysqld,
+		DataVolumePVCName:        key.Name,
+		DataVolumePVCSpec:        pool.DataVolumeClaimTemplate,
+		KeyspaceName:             keyspaceName,
+		DatabaseName:             vts.Spec.DatabaseName,
+		DatabaseInitScriptSecret: vts.Spec.DatabaseInitScriptSecret,
+		BackupLocation:           backupLocation,
+		BackupEngine:             vts.Spec.BackupEngine,
+		InitContainers:           pool.InitContainers,
+		SidecarContainers:        pool.SidecarContainers,
+		ExtraEnv:                 pool.ExtraEnv,
+		Annotations:              vts.Spec.Annotations,
+	}
+	update.Annotations(&tabletSpec.Annotations, backupLocation.Annotations)
```

Comment:
We should also do this for vttablet Pods, since they also access backup storage.

Reply:
Done.